manifests/startup.pp

```diff
@@ -11,6 +11,7 @@
 # @param service_type Systemd service type
 # @param manage_database When true, it will configure the database and execute the sql scripts.
 # @param service_name Name of the service. Defaults to the resource name
+# @param binary_location This params is use for define a specific binary location. This is actually only available for zabbix agent and systemd
```

Comment:
do we need the whole startup.pp for the new agent? doesn't it ship with a ready to use unit file?

Reply:
Ok, I've checked the deb package and both zabbix-agent and zabbix-agent2 supply a system-unit
So, probably is needed to remove the unit dropin built in the startup class or add a conditional parameter to use it if true and default to false

What do you think is better?

Reply:
I think we should not touch the existing setup for the old agent. That would become complicated. But if the new agent ships a unit, we should try to use it.

Reply:
Ok, I't now use the code shipped with the module adding just the way to change the path, so, probably I can add a condition using agent2 flag to skip unit creation

Reply:
it seems to fail on non debian os, check it deeply

Reply:
So, different OS use different location for pid in the unit file
what you think is the best way to approch this problem?

making custom unit mask it